### ISSUE TITLE
"Relevance" order of products should not be random

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#18605]
#### Describe the bug

When "Relevance" (or I don't know the translation) order is selected, the order of products is random. Sometimes, you go to second or third page in the category and see the product you already saw on previous pages.

#### Expected behavior

Ordering should not be completely random.

#### Additional information

* PrestaShop version: 1.7.6.2
* PHP version: 7.2

**Comment by MatShir** [comment]
Could you please provide us with more info? Are you talking about "Relevance" sorting in the category page? What is your suggestion? How you will like to see it sorted?

I can't find in the doc the way "relevance" sort the product. Do you have more insight ? @marionf @PrestaShop/prestashop-core-developers 

Thank you!

**Comment by Hlavtox** [comment]
@MatShir Exactly, thanks for specifying it.

When you select this sorting as default and go to some category. Hit F5 a few times and see the order of product change. It changes with every refresh and what sometimes happens is, that you see one single products on MULTIPLE pages.

I think EPIC should be made for the sorting options.
- My PR https://github.com/PrestaShop/PrestaShop/issues/14244 was marked as duplicate of https://github.com/PrestaShop/PrestaShop/issues/14210, but it was not fixed properly. There are different sorting options in Prestashop BO, FO category, FO best sales, FO new products.
- https://github.com/PrestaShop/PrestaShop/issues/17136 - I think the "bestselling" order is the most important of them, for a general customer. And its not available in category.

What do you think? 🚀

**Comment by marionf** [comment]
@MatShir 

The "relevance" option in category listing depends on what you have selected in Shop parameters > products: default order by & default order method

![capture d'écran_2523](https://user-images.githubusercontent.com/13449658/79367327-f94cf100-7f4d-11ea-8d88-2a710c996fd0.png)

![capture d'écran_2524](https://user-images.githubusercontent.com/13449658/79367332-fc47e180-7f4d-11ea-9c9a-8f945f60f1d6.png)

@Hlavtox 

> #17136 - I think the "bestselling" order is the most important of them, for a general customer. And its not available in category.

It seems another subject, let's stay focus on your issue here

> Hit F5 a few times and see the order of product change. It changes with every refresh and what sometimes happens is, that you see one single products on MULTIPLE pages.

To reproduce your issue we need more informations:
Could you please:
- add a screenrecord of the issue when you hit F5 few tiimes, 
- a screenshot of your configuration in Shop parameters > products: default order by & default order method 
- and tell us if you use facetedsearch module and in which version ?

**Comment by Hlavtox** [comment]
@marionf @MatShir 
- I am using Faceted search 3.4.0
- Product sorting by: Quantity in stock, Product sorting order: Descending ...... this settings shows up as "Relevance" in front office. (#14244)

**Live example**
Its perfectly visible in this category in our store. Watch the black glass sink, it will appear on both pages. Try it several times and you will definitely see it.
https://eshop-teka.cz/476-drezy-s-odkapem

**Some recorded cases**
I set number of products on the page to 6, to better see it. You can see that some of the products are not even visible in some passes... :(

![try2](https://user-images.githubusercontent.com/6097524/79374811-edfec300-7f57-11ea-8e75-b7cc210e9a61.png)

![try1](https://user-images.githubusercontent.com/6097524/79374805-eccd9600-7f57-11ea-96e7-76491e50c58c.png)

**Comment by khouloudbelguith** [comment]
Hi @Hlavtox,

I did not manage to reproduce the issue with PS1.7.6.4 and ps_facetedsearch v3.5.0.
I attached a screen record
https://drive.google.com/file/d/1Mn0nY4YkxG3YyFQCtTbyDgpEx0kWCxDi/view?usp=sharing

Thanks to check and feedback.

**Comment by Hlavtox** [comment]
@khouloudbelguith 

I saw you have the positions set in the category. I don't have any position set on any product, except few manually added ones. I import everything from CSV.

![Výstřižek](https://user-images.githubusercontent.com/6097524/79487677-ebb06d80-8018-11ea-8e7a-35876a862796.JPG)

I think maybe a fallback order should be implemented, to prevent these kinds of issue? What is the exact orderig of "Relevance" order? Quantity, then Position? If the position was the same, it could order by name or ID, so it didn't do these issues.

**UPDATE: Product positions have nothing to do with it, I have set them up and still the same :(**
